OpenRA.Mods.Common/UpdateRules/Rules/DecoupleSelfReloading.cs

```diff
@@ -0,0 +1,83 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+
+namespace OpenRA.Mods.Common.UpdateRules.Rules
+{
+	public class DecoupleSelfReloading : UpdateRule
+	{
+		public override string Name { get { return "Replace 'SelfReloads' with 'ReloadAmmoPool'"; } }
+		public override string Description
+		{
+			get
+			{
+				return "'SelfReloads', 'SelfReloadDelay', 'ReloadCount', 'ResetOnFire'\n" +
+					"and 'RearmSound' were renamed and moved to a new 'ReloadAmmoPool' trait.";
+			}
+		}
+
+		public override IEnumerable<string> UpdateActorNode(ModData modData, MiniYamlNode actorNode)
+		{
+			var poolNumber = 0;
+			var ammoPools = actorNode.ChildrenMatching("AmmoPool");
+			foreach (var pool in ammoPools)
+			{
+				var selfReloads = pool.LastChildMatching("SelfReloads");
+				if (selfReloads == null || selfReloads.NodeValue<bool>())
+					continue;
+
+				poolNumber++;
+				var reloadOnCond = new MiniYamlNode("ReloadAmmoPool@" + poolNumber, "");
+
+				var name = pool.LastChildMatching("Name");
+				if (name != null)
+					reloadOnCond.AddNode("AmmoPool", name.NodeValue<string>());
+
+				var selfReloadDelay = pool.LastChildMatching("SelfReloadDelay");
+				if (selfReloadDelay != null)
+				{
+					selfReloadDelay.RenameKeyPreservingSuffix("Delay");
+					reloadOnCond.AddNode(selfReloadDelay);
+					pool.RemoveNodes("SelfReloads");
```

Comment:
Why do you only remove "SelfReloads" if `selfReloadDelay != null`? It might have been using the internal default. I'd put this right after the 'continue' in the beginning of the loop, as we don't need it afterwards.

Reply:
I copied that from the original rule. Moved it out, though.